services/accounts/service.go

```diff
@@ -14,8 +17,23 @@ import (
 )
 
 // NewService initializes service instance.
-func NewService(db *accounts.Database, mdb *multiaccounts.Database, manager *account.GethManager, config *params.NodeConfig, feed *event.Feed, mediaServer *server.MediaServer) *Service {
-	return &Service{db, mdb, manager, config, feed, nil, mediaServer}
+func NewService(db *accounts.Database, mdb *multiaccounts.Database, manager *account.GethManager, config *params.NodeConfig, feed *event.Feed, publisher *pubsub.Publisher, mediaServer *server.MediaServer) *Service {
```

Comment:
Instead of passing the publisher to the services, maybe the services should create one themselves? 
Then instead of passing the publisher around everywhere, we can pass specific subscriptions/channels?

Reply:
Something like in `Manager`, but also not allow to `GetPublisher`, but only `Subscribe`

Reply:
YES, my vision for this is that whatever provides some type of data should instantiate a Publisher as well. That kind of encourages usage of light event structs, since you've got the data source next to the place where you perform the subscription. I didn't do it here due to
```
Commits after the first one replace uses of event.Feed in status-go with the new pubsub util. Only a direct replacement was meant, there's a bunch of rework potential with regards to how Publishers are passed around.
```